modules/local/sra_fastq_ftp.nf

```diff
@@ -31,17 +32,17 @@ process SRA_FASTQ_FTP {
     script:    
     if (meta.single_end) {
         """
-        curl -L ${fastq[0]} -o ${meta.id}.fastq.gz
+        timeout $task.time bash -c 'until curl $args -L ${fastq[0]} -o ${meta.id}.fastq.gz; do sleep 1; done'; echo -e \\\\a
         echo "${meta.md5_1} ${meta.id}.fastq.gz" > ${meta.id}.fastq.gz.md5
         md5sum -c ${meta.id}.fastq.gz.md5
         """
     } else {
         """
-        curl -L ${fastq[0]} -o ${meta.id}_1.fastq.gz
+        timeout $task.time bash -c 'until curl $args -L ${fastq[0]} -o ${meta.id}_1.fastq.gz; do sleep 1; done'; echo -e \\\\a

```

Comment:
Just need to make sure that `task.time` gets evaluated to the correct format as expected by `timeout`?

Reply:
It does indeed 🎉 `timeout 6h bash -c `

Reply:
yeah i saw this, phew!